pootle/apps/pootle_app/management/commands/update_tmserver.py

```diff
@@ -86,63 +184,66 @@ def _parse_translations(self, **options):
 
         self.stdout.write("")
 
-        for i, unit in enumerate(units_qs.iterator(), start=1):
-            fullname = (unit['submitted_by__full_name'] or
-                        unit['submitted_by__username'])
-            project = unit['store__translation_project__project__fullname']
-
-            email_md5 = None
-            if unit['submitted_by__email']:
-                email_md5 = md5(unit['submitted_by__email']).hexdigest()
-
+        for i, unit in enumerate(units, start=1):
             if (i % 1000 == 0) or (i == total):
                 percent = "%.1f" % (i * 100.0 / total)
                 self.stdout.write("%s (%s%%)" % (i, percent), ending='\r')
                 self.stdout.flush()
 
-            yield {
-                "_index": self.INDEX_NAME,
-                "_type": unit['store__translation_project__language__code'],
-                "_id": unit['id'],
-                'revision': int(unit['revision']),
-                'project': project,
-                'path': unit['store__pootle_path'],
-                'username': unit['submitted_by__username'],
-                'fullname': fullname,
-                'email_md5': email_md5,
-                'source': unit['source_f'],
-                'target': unit['target_f'],
-            }
+            yield self.parser.get_unit_data(unit)
 
         if i != total:
             self.stdout.write("Expected %d, loaded %d." % (total, i))
 
-
-    def handle(self, *args, **options):
+    def _get_settings(self, *args, **options):
```

Comment:
This method doesn't return anything (an implicit `None` aside); it doesn't make sense to call it as such (`_get_...`)